src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,221 @@
+            arguments = arguments.substring(0, arguments.length() - 1);
+        }
+
+        switch (action) {
+        case AddCommand.COMMAND_WORD_D:
+        case AddCommand.COMMAND_WORD_E:
+        case AddCommand.COMMAND_WORD_T:
+            return handleAdd(action, arguments, marked);
+        case DeleteCommand.COMMAND_WORD:
+            if (arguments.equals("")) {
+                throw new InvalidArgumentException();
+            }
+            return new DeleteCommand(parseArgs(arguments));
+        case ListCommand.COMMAND_WORD:
+            if (!arguments.equals("")) {
+                throw new InvalidCommandException();
+            }
+            return new ListCommand();
+        case ExitCommand.COMMAND_WORD:
+            return new ExitCommand();
+        case MarkCommand.COMMAND_WORD_UNMARK:
+            if (arguments.equals("")) {
+                throw new InvalidArgumentException();
+            }
+            return new MarkCommand(false, parseArgs(arguments));
+        case MarkCommand.COMMAND_WORD_MARK:
+            if (arguments.equals("")) {
+                throw new InvalidArgumentException();
+            }
+            return new MarkCommand(true, parseArgs(arguments));
+        case FindCommand.COMMAND_WORD:
+            if (arguments.equals("")) {
+                throw new InvalidArgumentException();
+            }
+            return new FindCommand(arguments);
+        default:
+            throw new InvalidCommandException();
+        }
+    }
+
+    /**
+     * Parses user input for adding into tasks.
+     *
+     * @param taskInput the type of Task
+     * @param args the arguments to be parsed for the Task
+     * @param isMarked boolean value if the Task is marked
+     * @return AddCommand to be executed
+     * @throws DukeException if there is invalid user input
+     */
+    protected static AddCommand handleAdd(String taskInput, String args, boolean isMarked) throws DukeException {
+        if (args.equals("")) {
+            throw new InvalidArgumentException();
+        }
+        Task task;
+        switch (taskInput) {
+        case AddCommand.COMMAND_WORD_D:
+            task = parseDeadline(args, isMarked);
+            break;
+        case AddCommand.COMMAND_WORD_E:
+            task = parseEvent(args, isMarked);
+            break;
+        case AddCommand.COMMAND_WORD_T:
+            task = new Todo(args, isMarked);
+            break;
+        default:
+            throw new InvalidCommandException();
+        }
+
+        return new AddCommand(task);
+    }
+
+    /**
+     * Parses user text into the action type and the arguments
+     *
+     * @param text the text to be parsed
+     * @return String[] where the first index is the action type and second index is the arguments
+     */
+    protected static String[] parseText(String text) {
+        String[] words = text.trim().split(" ");
+        String[] remaining = Arrays.copyOfRange(words, 1, words.length);
```

Comment:
What do you think of a variable name with a plural s, such as `remainingWords`?

Reply:
Thanks for the suggestion